src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -3385,22 +3385,30 @@ void MergeTreeData::checkAlterPartitionIsPossible(
             else
             {
                 /// We are able to parse it
-                getPartitionIDFromQuery(command.partition, getContext());
+                const auto * partition_ast = command.partition->as<ASTPartition>();
+                if (partition_ast &&  partition_ast->all)
```

Comment:
```suggestion
                if (partition_ast && partition_ast->all)
```

Reply:
I agree with your suggestion and have fix it.